File: Progamming_Assignments/PHP/References/TLoop.php
<html>
   <head>
     <title>PHP Test Loops </title>
   </head>
   <body>
      <?php
         // Test Loops: while, for, nested loops (1-D, 2-D, and Associative Arrays)
         // Program-ID: TLoop.php
         // Author:     Kuo-pao Yang
         // OS:         Ubuntu 24
         // Parser:     PHP 8

         $a = array(1, 2, 3);
         $b = array( array(10, 20, 30),
                     array(40, 50, 60),
                     array(70, 80, 90));
         //Associative Arrays
         $c = array("p"=>100, "q"=>200, "r"=>300);
         $d = array("x"=>array(100, 200, 300),
                    "y"=>array(400, 500, 600),
                    "z"=>array(700, 800, 900));
 
         print "<br>Test while loop: 1-D Array<br>";
         $i = 0;
         while($i < 3) {
            print "a[".$i."]=".$a[$i]." ";
            $i++;
         }

         print "<br>Test for loop: 2-D Array<br>";
         for($j = 0; $j < 3; $j++) {
            print "b[1,".$j."]=".$b[1][$j]." ";
         }

         print "<br>Test nested loop: 2-D Array";
         for($i = 0; $i < 3; $i++) {
            print "<br>";
            for($j = 0; $j < 3; $j++) {
               print "b[".$i.",".$j."]=".$b[$i][$j]." ";
            }
         }

         print "<br>Test Associative 1-D and 2-D Arrays<br>";
         foreach ($c as $e => $f)
            print "c[".$e."]=".$f." ";
         foreach ($d as $e => $f) {
            print"<br>";
            for ($j = 0; $j < 3; $j++)
               print "d[".$e.",".$j."]=".$f[$j]." ";
         }
      
         /* Output:
         Test while loop: 1-D Array
         a[0]=1 a[1]=2 a[2]=3 
         Test for loop: 2-D Array
         b[1,0]=40 b[1,1]=50 b[1,2]=60 
         Test nested loop: 2-D Array
         b[0,0]=10 b[0,1]=20 b[0,2]=30 
         b[1,0]=40 b[1,1]=50 b[1,2]=60 
         b[2,0]=70 b[2,1]=80 b[2,2]=90 
         Test Associative 1-D and 2-D Arrays
         c[p]=100 c[q]=200 c[r]=300 
         d[x,0]=100 d[x,1]=200 d[x,2]=300 
         d[y,0]=400 d[y,1]=500 d[y,2]=600 
         d[z,0]=700 d[z,1]=800 d[z,2]=900 
         */
      ?> 
   </body>
</html> 

File: Progamming_Assignments/C/Sample_Code/TLoop.c
// Test Loop: while, for, nested loops (1-D Arrary and 2-D Array)
// Program-ID:   TLoop.c
// Author:       Kuo-pao Yang
// OS:           Ubuntu 24
// Compiler:     GNU C
// Note:
// The following instructions are used to
//      edit, compile, and run this program
//    $ nano     TLoop.c
//    $ gcc      TLoop.c 
//    $ ./a.out

#include <stdio.h>

void main() {
    int i, j;
    int a[3] = {1, 2, 3};
    int b[3][3] = {{10, 20, 30},
                   {40, 50, 60},
                   {70, 80, 90}};
    int *p;

    printf("\nTest while loop: 1-D Array and Pointer\n");
    p = &a[0];
    i = 0;
    while(i < 3) {
       printf("a[%d] = %d, *p = %d\t", i, a[i], *p);
       p++;
       i++;
    }

    printf("\nTest for loop: 2-D Array and Pointer\n");
    p = &b[1][0];
    for(j = 0; j < 3; j++) {
        printf("b[1,%d] = %d, *p = %d\t", j, b[1][j], *p);
        p++;
    }

    printf("\nTest nested loop: 2-D Array and Pointer");
    p = &b[0][0]; 
    for(i = 0; i < 3; i++) {
      printf("\n");
      for(j = 0; j < 3; j++) {
        printf("b[%d,%d] = %d, *p = %d\t", i, j, b[i][j], *p);
        p++;
      }
    }
}

/* Output:
Test while loop: 1-D Array and Pointer
a[0] = 1, *p = 1        a[1] = 2, *p = 2        a[2] = 3, *p = 3
Test for loop: 2-D Array and Pointer
b[1,0] = 40, *p = 40    b[1,1] = 50, *p = 50    b[1,2] = 60, *p = 60
Test nested loop: 2-D Array and Pointer
b[0,0] = 10, *p = 10    b[0,1] = 20, *p = 20    b[0,2] = 30, *p = 30
b[1,0] = 40, *p = 40    b[1,1] = 50, *p = 50    b[1,2] = 60, *p = 60
b[2,0] = 70, *p = 70    b[2,1] = 80, *p = 80    b[2,2] = 90, *p = 90 
*/
